### PR TITLE
Add the `sv_in_set()` function

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR adds the `sv_in_set()` function. It's a validator function that checks whether the field value is part of a specified set of values.

Fixes: https://github.com/rstudio/shinyvalidate/issues/12

### CODE REVIEW

File: R/rules.R [hunk 1]
```diff
@@ -295,3 +294,93 @@ sv_between <- function(left,
...
+#' 
+#'   iv <- InputValidator$new()
+#' 
+#'   iv$add_rule("count", sv_in_set(1:5))
+#'   iv$add_rule("count", ~if (. <= 0) "A positive value is required")
+#'
+#' })
+#' @export
+sv_in_set <- function(set,
+                      allow_nan = FALSE,
```
Comment (jcheng5) [inline R/rules.R:333]: Let's lose `allow_nan`, people can just include `NaN` in the `set`

File: R/rules.R [hunk 2]
```diff
@@ -295,3 +294,93 @@ sv_between <- function(left,
     }
   }
 }
+
+#' Validate that a field is part of a defined set
+#'
+#' The `sv_in_set()` function checks whether the field value is part of a
+#' specified set of values.
+#' 
+#' @param set A vector of elements for which the field value must be a part of
```
Comment (jcheng5) [inline R/rules.R:303]: "A vector or list of elements"
Also note that NA and NaN can be included if desired

File: R/rules.R [hunk 3]
```diff
@@ -295,3 +294,93 @@ sv_between <- function(left,
...
+#'   to pass validation. To allow an empty field, `NA` should be included in the
+#'   `set` vector.
+#' @param allow_nan If `FALSE` (the default), then the presence of a `NaN` value
+#'   element will cause validation to fail.
+#' @param message The validation error message to use if a value fails to match
+#'   the rule. By default, this is generic message provided by **shinyvalidate**
```
Comment (jcheng5) [inline R/rules.R:309]:
```suggestion
#'   the rule. By default, this is a generic message provided by **shinyvalidate**
```

File: R/rules.R [hunk 4]
```diff
@@ -295,3 +294,93 @@ sv_between <- function(left,
...
+#'   iv <- InputValidator$new()
+#' 
+#'   iv$add_rule("count", sv_in_set(1:5))
+#'   iv$add_rule("count", ~if (. <= 0) "A positive value is required")
+#'
+#' })
+#' @export
+sv_in_set <- function(set,
+                      allow_nan = FALSE,
+                      message = NULL,
```
Comment (jcheng5) [inline R/rules.R:334]: Let's let the user provide `{value}` token in their message

File: R/rules.R [hunk 5]
```diff
@@ -295,3 +294,93 @@ sv_between <- function(left,
...
+                      msg_limit = 3) {
+  force(allow_nan)
```
Comment (jcheng5) [inline R/rules.R:336]: `force(set)`

Comment (jcheng5) [reply R/rules.R:336]: Throw if `set` is empty

File: R/rules.R [hunk 6]
```diff
@@ -295,3 +294,93 @@ sv_between <- function(left,
...
+    
+    message <- 
+      glue::glue_data_safe(
+        list(values_text = values_text),
+        "Must be in the set of {values_text}."
+      )
+  }
+  
+  function(value) {
+
```
Comment (jcheng5) [inline R/rules.R:352]: `value` can have length != 1

File: R/rules.R [hunk 7]
```diff
@@ -295,3 +294,93 @@ sv_between <- function(left,
...
+      )
+  }
+  
+  function(value) {
+    
+    if (!allow_nan && any(is.nan(value))) {
+      return(message)
+    }
+    
+    if (!(value %in% set)) {
```
Comment (jcheng5) [inline R/rules.R:357]:
```suggestion
    if (!all(value %in% set)) {
```

File: R/rules.R (hunk 3 above)
Comment (rich-iannone) [reply R/rules.R:309]: Fully revised this paragraph.

File: R/rules.R (hunk 2 above)
Comment (rich-iannone) [reply R/rules.R:303]: Changed now.

File: R/rules.R (hunk 1 above)
Comment (rich-iannone) [reply R/rules.R:333]: Done.

File: R/rules.R (hunk 4 above)
Comment (rich-iannone) [reply R/rules.R:334]: Now done.

File: R/rules.R (hunk 5 above)
Comment (rich-iannone) [reply R/rules.R:336]: Both done now.

File: R/rules.R (hunk 7 above)
Comment (rich-iannone) [reply R/rules.R:357]: Changed now.

**Review by jcheng5 — APPROVED** [review]

**Pull request merged**